src/crimson/osd/osd_operations/client_request.h

```diff
@@ -74,7 +74,7 @@ class ClientRequest final : public OperationT<ClientRequest>,
   PGPipeline &pp(PG &pg);
 
   OpSequencer& sequencer;
-  uint64_t prev_op_id = 0;
+  std::optional<uint64_t> prev_op_id;
```

Comment:
instead of using an extra `bool` in `optional<>`,  has you considered using a magic number to represent the unset value

```c++
prev_op_id = UINT64_C(-1);
```
or 
```c++
prev_op_id = std::max<uint64_t>::max();
```

? probably it's but yet another attempt of trading readability for a premature optimization...

Reply:
`prev_op_id = std::max<uint64_t>::max()` would be preferred from these two. However, I'm not sure we really need to optimize this bit.

Anyway, what's the reason for the 2 stage construction? Do you recall what prohibited us from setting `prev_op_id` in the constructor?

Reply:
IIRC, i followed your suggestion to initialize `prev_op_id` in the ctor. but latter on i was forced to use the 2-stage construction.. i cannot recall why though... need to do to some archaeology.

Reply:
#40021

Reply:
Ah, of course! Thanks for digging this out!